gateway/src/service.rs

```diff
@@ -402,7 +398,41 @@ impl GatewayService {
         &self,
         project_name: ProjectName,
         account_name: AccountName,
-    ) -> Result<Work, Error> {
+    ) -> Result<Project, Error> {
+        if let Some(row) = query("SELECT project_name, account_name, initial_key, project_state FROM projects WHERE project_name = ?1 AND account_name = ?2")
+            .bind(&project_name)
+            .bind(&account_name)
+            .fetch_optional(&self.db)
+            .await?
+        {
+            // If the project already exists and belongs to this account
+            let project = row.get::<SqlxJson<Project>, _>("project_state").0;
+            if project.is_destroyed() {
+                // But is in `::Destroyed` state, recreate it
+                let initial_key = row.get::<String, _>("initial_key");
```

Comment:
Should we consider cycling the initial key here?

Reply:
No prob, will do.